flang/lib/Lower/IO.cpp

```diff
@@ -943,16 +943,28 @@ lowerReferenceAsStringSelect(
     auto eval = converter.lookupLabel(label);
     assert(eval && "Label is missing from the table");
 
-    auto stringLit = lowerSourceTextAsStringLit(
-        converter, loc, toStringRef(eval->position), strTy, lenTy);
-    auto stringRef = std::get<0>(stringLit);
-    auto stringLen = std::get<1>(stringLit);
+    auto text = toStringRef(eval->position);
+    mlir::Value stringRef;
+    mlir::Value stringLen;
+    if (text.find_lower("format") != llvm::StringRef::npos) {
```

Comment:
You may want to also check  `eval->isA<parser::FormatStmt>()` in this `if`, otherwise I think the following kind of code would probably still crash the compiler:

```
010 my_format_var = 10
       assign 10 to I
       write(*, I)
```

Reply:
I'll try that. Thanks.